type-definitions/most.d.ts

```diff
@@ -1,385 +1,788 @@
+		end(time: number, value?: T): void;
+		error(time: number, err: Error): void;
+	}
+
+	export default DeferredSink;
+}
+
+declare module "most/lib/sink/DeferredSink" {
+	import DeferredSink from 'most/src/sink/DeferredSink';
+	export = DeferredSink;
+}
+
+declare module "most/src/sink/IndexSink" {
+	import { Sink } from 'most';
+
+	class IndexSink<T> implements Sink<T> {
+		public index: number;
+		public value: T;
+		public active: boolean;
+		public sink: Sink<IndexSink<T>>;
+		constructor(index: number, sink: Sink<IndexSink<T>>);
+		event(time: number, value: T): void;
+		end(time: number, value?: T): void;
+		error(time: number, err: Error): void;
+	}
+
+	export default IndexSink;
+}
+
+declare module "most/lib/sink/IndexSink" {
+	import IndexSink from 'most/src/sink/IndexSink';
+	export = IndexSink;
+}
+
+declare module "most/src/sink/Pipe" {
+	import { Sink } from 'most'
+
+	class Pipe<T> implements Sink<T> {
+		constructor(sink: Sink<T>);
+		event(time: number, value: T): void;
+		end(time: number, value?: T): void;
+		error(time: number, err: Error): void;
+	}
+
+	export default Pipe;
+}
+
+declare module "most/lib/sink/Pipe" {
+	import Pipe from 'most/src/sink/Pipe';
+	export = Pipe;
+}
+
+declare module "most/src/sink/SafeSink" {
+	import { Sink } from 'most';
+
+	class SafeSink<T> implements Sink<T> {
+		public active: boolean;
+		constructor(sink: Sink<T>);
+		event(time: number, value: T): void;
+		end(time: number, value?: T): void;
+		error(time: number, err: Error): void;
+		disable(): Sink<T>;
+	}
+
+	export default SafeSink;
+}
+
+declare module "most/lib/sink/SafeSink" {
+	import SafeSink from 'most/src/sink/SafeSink';
+	export = SafeSink;
+}
+
+declare module "most/src/fatalError" {
+	function fatalError (err: Error): void;
+
+	export default fatalError;
+}
+
+declare module "most/lib/fatalError" {
+	import fatalError from 'most/src/fatalError';
+	export = fatalError;
+}
+
+declare module "most/src/invoke" {
+	interface InvokeFn {
+		<A, B>(f: (a: A) => B, args: [A]): B;
+		<A, B, C>(f: (a: A, b: B) => C, args: [A, B]): C;
+		<A, B, C, D>(f: (a: A, b: B, c: C) => D, args: [A, B, C]): D;
+		<A, B, C, D, E>(f: (a: A, b: B, c: C, d: D) => E, args: [A, B, C, D]): E;
+		<R>(f: (...args: any[]) => R, args: any[]): R;
+	}
+
+	const invoke: InvokeFn;
+
+	export default invoke;
+}
+
+declare module "most/lib/invoke" {
+	import invoke from "most/src/invoke"
+	export = invoke;
+}
+
+declare module "most/src/iterable" {
+	// es5 doesn't have typings for an Iterable
+	// use any to allow type-casting
+	type Iterable = any;
+
+	export function isIterable(x: any): boolean;
+	export function getIterator(x: any): Iterable;
+	export function makeIterable(f: Function, obj: any): Iterable;
+}
+
+declare module "most/lib/iterable" {
+	export * from 'most/src/iterable';
+}
+
+declare module "most/src/LinkedList" {
+	class LinkedList<T> {
+		add (x: T): void;
+		remove (x: T): void;
+		isEmpty(): boolean;
+		dispose(): Promise<T>;
```

Comment:
This doesn't seem quite right. The implementation can return a Promise with a resolved value of `nothing` or an array.